### PR TITLE
Main app

### LINKED ISSUE #32: Project structure refactor
We need to refactor the apps inside the django project so it will contain each logic inside its own app.
The first app we have wrote has a lot of unnecessary code and does not work work static assets and base templates.

### PR DESCRIPTION
**Pull request by or313**
This PR complete the work on #32 which is adding the main app and base template for the project.
This main app will eventually include all the features which we intend to develop
fix #32

### CODE REVIEW

**Review by lmilbaum — CHANGES_REQUESTED** [review]
Same request as in #36 
Link an issue and a descriptive target

**Comment by ravidbro** [comment]
The branch has conflicts, please rebase and fix the conflicts

**Comment by ravidbro** [comment]
also, the issue doesn't seem to be linked (in the right pane), use the word 'fix'.
you can see an example in https://github.com/beyond-io/StuDash/pull/38

File: StuDash/settings.py [hunk 1]
```diff
@@ -1,108 +1,140 @@
...
-    'django.contrib.sessions',
-    'django.contrib.messages',
-    'django.contrib.staticfiles',
+    "django.contrib.admin",
+    "django.contrib.auth",
+    "django.contrib.contenttypes",
+    "django.contrib.sessions",
+    "django.contrib.messages",
+    "django.contrib.staticfiles",
+    "django.contrib.sites",
```
Comment (simzacks) [inline StuDash/settings.py:26]: django.contrib.sites is for hosting multiple sites with one backend. Are you planning on doing this? if not, do you have another reason for using this?

File: StuDash/settings.py [hunk 2]
```diff
@@ -1,108 +1,140 @@
...
+    # Vendor apps
+    "bootstrap4",
+    # Application apps
+    "main",
```
Comment (simzacks) [inline StuDash/settings.py:30]: you might want to call your application studash, instead of "main"

File: StuDash/settings.py [hunk 3]
```diff
@@ -1,108 +1,140 @@
...
+
+SIGN_UP_FIELDS = [
+    "username",
+    "first_name",
+    "last_name",
+    "email",
+    "password1",
+    "password2",
+]
+if DISABLE_USERNAME:
```
Comment (simzacks) [inline StuDash/settings.py:115]: This doesn't look like a setting that you will change back and forth. This is the type of setting that is used for applications with multiple installations that you want to give the sysadmin some options. However, for a small project like this it is very extra.
Try to develop minimally.

File: content/assets/js/ie10-viewport-bug-workaround.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
+ */
+
+// See the Getting Started docs for more information:
+// https://getbootstrap.com/getting-started/#support-ie10-width
+
+(function () {
+  'use strict'
+
+  if (navigator.userAgent.match(/IEMobile\/10\.0/)) {
```
Comment (simzacks) [inline content/assets/js/ie10-viewport-bug-workaround.js:14]: I would suggest that for the MVP you make sure it works with one browser and only write code that will work for that. In the future if you want to expand you can verify the GUI on multiple browsers.

File: content/templates/main/change_language.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{% extends 'layouts/default/page.html' %}
+
+{% load i18n %}
+
+{% block content %}
+
+    {% get_current_language as language_code %}
+    {% get_available_languages as languages %}
+
+    <h4>{% trans 'Change language' %}</h4>
```
Comment (simzacks) [inline content/templates/main/change_language.html:10]: Are multiple languages part of your MVP? That is generally considered an advanced feature.

File: content/templates/layouts/default/page.html [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    <title>Student dashboard for learing from home</title>
+
+    <link
+      rel="stylesheet"
+      href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css"
+      integrity="sha384-9aIt2nRpC12Uk9gS9baDl411NQApFmC26EwAOH8WgZl5MYYxFfc+NcPb1dKGj7Sk"
+      crossorigin="anonymous"
+    />
+
+    <link href="{% static 'css/styles.css' %}" rel="stylesheet" />
```
Comment (simzacks) [inline content/templates/layouts/default/page.html:28]: Your (empty) css file is not in the content/static/css folder. Will this work?

**Review by simzacks — COMMENTED** [review]
I recommend that you add something to the css file before uploading it. Having a blank file that _may be_ used in the future is bad form.

**Review by ravidbro — APPROVED** [review]

**Pull request merged**